manifests/master/gitea.pp

```diff
@@ -1,17 +1,53 @@
 # This assumes that the gitea rpm and dependencies have been cached by the
 # pltraining-bootstrap module.
 class classroom::master::gitea {
+
+  # Use public IP for ec2 hosted trainings
+  $gitea_address = $facts['ec2_metadata'] ? {
+        undef   => $ipaddress,
+        default => $facts['ec2_metadata']['public-ipv4']
+  }
+
   package { 'gitea':
     ensure => present,
-    before => File['/home/git/go/bin/custom/conf/app.ini'],
   }
 
-  file { '/home/git/go/bin/custom/conf/app.ini':
-    ensure  => file,
-    owner   => 'git',
-    group   => 'git',
-    mode    => '0644',
-    source  => 'puppet:///modules/classroom/app.ini',
+  ini_setting { "database user":
+    ensure  => present,
+    path    => '/home/git/go/bin/custom/conf/app.ini',
+    section => 'database',
+    setting => 'user',
+    value   => 'root',
+    require => Package['gitea'],
+    notify  => Service['gitea'],
+  }
+
+  ini_setting { "ssh domain":
+    ensure  => present,
+    path    => '/home/git/go/bin/custom/conf/app.ini',
+    section => 'server',
+    setting => 'SSH_DOMAIN',
+    value => $gitea_address,
+    require => Package['gitea'],
+    notify  => Service['gitea'],
+  }
+
+  ini_setting { "domain":
+    ensure  => present,
+    path    => '/home/git/go/bin/custom/conf/app.ini',
+    section => 'server',
+    setting => 'DOMAIN',
+    value => $gitea_address,
+    require => Package['gitea'],
+    notify  => Service['gitea'],
+  }
+  ini_setting { "root url":
```

Comment:
I found these by doing a side by side comparison between the default file and one from a working installation. So I think this should be it.